### PR TITLE
Add "http" as Jest Puppeteer server protocol and add --runInBand arg to Jest Puppeteer commands

### PR DESCRIPTION
**Pull request by mjlumetta**
### Description

does a handful of things for [API-2787](https://vajira.max.gov/browse/API-2787). 

the most important is that it explicitly adds `protocol: 'http'` to the Jest Puppeteer dev server config (https://github.com/smooth-code/jest-puppeteer/tree/master/packages/jest-dev-server#protocol). `port` and `protocol` tell Jest Puppeteer to wait for a connection at that protocol and port on localhost. in [4.2.0](https://github.com/smooth-code/jest-puppeteer/blob/master/CHANGELOG.md#420-2019-05-28), Jest Puppeteer changed from defaulting to HTTP to defaulting to TCP in [this commit](https://github.com/smooth-code/jest-puppeteer/commit/8b64c10). since the URL changed, i imagine there were some issues checking that the server was running. i'm not sure why it only fails some of the time (or maybe it fails most of the time with the 30s timeout?), and i'm not sure exactly what the cases here are. going to continue looking into the cause and trying to get something more definite.

there are two other things in this PR:

- i added the `--runInBand` flag to the CI commands and will probably push a commit to add it to the non-CI commands since we usually run them in Docker. (although i don't necessarily want to _assume_ that about Docker, and Docker users can always do `npm run test:e2e -- --runInBand`, so maybe not.) anyway, the background is here: https://jestjs.io/docs/en/troubleshooting#tests-are-extremely-slow-on-docker-andor-continuous-integration-ci-server
- i added an environment variable to the Makefile that allows devs to specify a custom image name for the Docker image.

one thing is that i'm haven't established complete and total evidence for why this fixes the issues. more research to come, hopefully, but i figure getting this PR up is worth it. also going to try to get some stats on Jest Puppeteer run times for comparison from CodeBuild or Jenkins.

### Process

- [n/a] Tests added or updated for change
  - tests do work more frequently now (dunno under what circumstances they might not)
- [n/a] Docs updated
- [n/a] Accessibility concerns have been considered and addressed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
     "test:unit": "node scripts/test.js --projects=unit.jest.config.js",
     "test:unit:ci": "node scripts/test.js --projects=unit.jest.config.js --ci --testResultsProcessor=\"./node_modules/jest-junit-reporter\"",
```
Comment (rtravitz) [inline package.json:12]: Would it also be an improvement to run the unit tests in band?

Comment (mjlumetta) [reply package.json:12]: that's a good question. let me see if i can grab some stats on the e2e and accessibility tests and then i'll look into the unit tests. i suspect the answer is yes but also want to back up the hypothesis about the single-process command for the Jest Puppeteer tests before we do that.

**Comment by mjlumetta** [comment]
recording some times of Jest runs using the Jest CLI's output

### This Branch

| e2e, CodeBuild | e2e, Jenkins | accessibility, CodeBuild | accessibility, Jenkins |
| ---------------------- | --------------------- | -------------------------- | ------------------------ |
| [22.15 s](https://console.amazonaws-us-gov.com/codesuite/codebuild/008577686731/projects/developer-portal-ci/build/developer-portal-ci%3Af9ac98e2-f39a-476f-b5af-af9be6b531d1/?region=us-gov-west-1) | [21.36 s](http://jenkins.vfs.va.gov/job/testing/job/developer-portal/job/API-2787%252Fjest-puppeteer-timeouts-6/1/console) | [49.86 s](https://console.amazonaws-us-gov.com/codesuite/codebuild/008577686731/projects/developer-portal-ci/build/developer-portal-ci%3Af9ac98e2-f39a-476f-b5af-af9be6b531d1/?region=us-gov-west-1) | [49.33 s](http://jenkins.vfs.va.gov/job/testing/job/developer-portal/job/API-2787%252Fjest-puppeteer-timeouts-6/1/console) |


### Other Branches

**Review by ediiotero — DISMISSED** [review]
Looks good. Running it locally and getting a few warnings from SupportFAQ

**Comment by mjlumetta** [comment]
same recording as above but for unit tests with `--runInBand` arg

### This Branch

| Commit | unit, CodeBuild | unit, Jenkins |
| -------- | --------------- | ------------- |
| 58354135d1027d05857670714db63fb99b4a4c1a | [34.96 s](https://console.amazonaws-us-gov.com/codesuite/codebuild/008577686731/projects/developer-portal-ci/build/developer-portal-ci%3A73eed474-3d46-45b7-8250-36045365c67f/?region=us-gov-west-1) | [31.60 s](http://jenkins.vfs.va.gov/job/testing/job/developer-portal/job/API-2787%252Fjest-puppeteer-timeouts-6/3/consoleFull) |


### Other Branches

| Commit | unit, CodeBuild | unit, Jenkins |
| -------- | --------------- | ------------- |

**Comment by mjlumetta** [comment]
recording some metrics for non-CI Jest Puppeteer commands to see if `--runInBand` improves `test:e2e` and `test:accessibility`

### e2e

| this branch, --runInBand, host | master, no --runInBand, host | this branch, --runInBand, Docker | master, no --runInBand, Docker |
| -------------------------- | ---------------------- | ----------------------------- | ------------------------ |
| 18.63 s | 29.21 s | 33.98 s | 105.43 s **FAILED** |
| 20.59 s | 26.67 s | 32.10 s | 102.77 s **FAILED** |
| 17.12 s | 27.63 s | 32.29 s | 101.86 s **FAILED** |
| 16.98 s | 26.26 s | 31.79 s | 99.38 s **FAILED** |
| 18.26 s | 28.22 s | 31.84 s | 101.96 s **FAILED** |

### accessibility

| this branch, --runInBand, host | master, no --runInBand, host | this branch, --runInBand, Docker | master, no --runInBand, Docker |
| -------------------------- | ---------------------- | ----------------------------- | ------------------------ |

**Review by gileswells — APPROVED** [review]
Overall looks good to me. 

I wonder if there's at all a benefit of leaving some slop in the timeout of something like 45 seconds vs the default 30 once removed. This gives us a little wiggle room but then we'll know that if we ever hit that then we've caused some sort of larger issue. Leaving at 30 where it'll sometimes spastically fail seems like false positives where a 45 second timeout would only fail on signs of a larger issue. I'm ok leaving it at 30 if you disagree.

**Comment by mjlumetta** [comment]
> Leaving at 30 where it'll sometimes spastically fail seems like false positives where a 45 second timeout would only fail on signs of a larger issue.

@gileswells for this bit - are you seeing intermittent failures? i haven't been seeing them with this current revision.

**Review by TS000 — APPROVED** [review]
LGTM, nice tables!

**Pull request merged**